src/components/Carousel.tsx

```diff
@@ -38,32 +57,47 @@ function Carousel(props: CarouselProps): JSX.Element {
         <button
           className={'carousel-btn prev'}
           style={{
-            visibility: ((props.showPrev !== undefined && props.showPrev) || (slideIdx > 0))
+            visibility: (isCurSlideValid() &&
+              ((props.children[slideIdx].showPrev === undefined &&
+                ((props.showPrev !== undefined && props.showPrev) || slideIdx > 0)) ||
+                (props.children[slideIdx].showPrev !== undefined && props.children[slideIdx].showPrev)))
               ? 'visible'
               : 'hidden',
           }}
           onClick={() => {
-            setSlideIdx(old => Math.max(old - 1, 0));
-            props.onPrev && props.onPrev();
+            goPrevSlide();
           }}
         >
           <img src={PrevSvg} />
         </button>
         <div id={'carousel-content'}>
-          {props.children && props.children.length > 0 && props.children[slideIdx]}
+          {
+            isCurSlideValid() &&
+            <div>
+              {props.children[slideIdx].topText && <h2 id={'body-text'}> {props.children[slideIdx].topText} </h2>}
+              <CarouselItem goNextSlide={function () { return goNextSlide(slideIdx); }}
+                goPrevSlide={function () { return goPrevSlide(slideIdx); }}
+                timeout={props.children[slideIdx].animationTime}>
+                {props.children[slideIdx].child}
+              </CarouselItem>
+              {props.children[slideIdx].bottomText && <h2 id={'body-text'}> {props.children[slideIdx].bottomText} </h2>}
+            </div>
```

Comment:
nit: Since `CarouselItem` is a sort of container for each "slide", I think it would be appropriate to have the `CarouselItem` contain the top and bottom text inside it. Something like this:
```suggestion
            <CarouselItem goNextSlide={function () { return goNextSlide(slideIdx); }}
              goPrevSlide={function () { return goPrevSlide(slideIdx); }}
              timeout={props.children[slideIdx].animationTime}>
              <div>
                {props.children[slideIdx].topText && <h2 id={'body-text'}> {props.children[slideIdx].topText} </h2>}
                {props.children[slideIdx].child}
                {props.children[slideIdx].bottomText && <h2 id={'body-text'}> {props.children[slideIdx].bottomText} </h2>}
              </div>
            </CarouselItem>
```
(note: I think this rearrangement shouldn't break alignment and stuff, but maybe double check on your end to be sure)

Reply:
+1 this

i also have been seeing the `props.children[slideIdx]` a lot.. can we make this a state variable?

```ts
const [child, setChild] = useState(props.children[slideIdx]);

useEffect(() => {
  setChild(props.children[slideIdx]);
}, [slideIdx]);

```

Reply:
hmm since its a one to one render (for every change in slideIdx, child also rerenders once) , i don't quite see why child should be a state variable? 

it could just be a temporary value, like `child = props.children[slideIdx]`?